Common/AR/source/ArcGISArViewInterface.cpp

```diff
@@ -201,42 +201,46 @@ LocationDataSource* ArcGISArViewInterface::locationDataSource() const
 }
 
 /*!
- * \brief Sets the location data source to \a locationDataSource.
- *
- * If \a locationDataSource is \c nullptr, the tracking of the LocationDataSource is disabled.
- * If \l tryUsingArKit is \c true, the AR framework is used for the tracking, not \l LocationDataSource.
+  \brief Sets the location data source to \a locationDataSource.
+
+  If \a locationDataSource is \c nullptr, the tracking of the LocationDataSource is disabled.
+  If \l tryUsingArKit is \c true, the AR framework is used for the tracking, not \l LocationDataSource.
  */
 void ArcGISArViewInterface::setLocationDataSource(LocationDataSource* locationDataSource)
 {
   if (m_locationDataSource == locationDataSource)
     return;
 
   m_locationDataSource = locationDataSource;
+  connect(m_locationDataSource, &LocationDataSource::locationChanged, this, &ArcGISArViewInterface::setLocationInternal);
+  connect(m_locationDataSource, &LocationDataSource::headingChanged, this, &ArcGISArViewInterface::setHeadingInternal);
+  emit locationDataSourceChanged();
+}
 
-  // update connection
-  disconnect(m_locationChangedConnection);
-  disconnect(m_headingChangedConnection);
+/*!
+  \brief The tracking mode controlling how the locations generated from the location data source
+  are used during AR tracking.
 
-  if (m_locationDataSource)
-  {
-    m_locationChangedConnection = connect(m_locationDataSource, &LocationDataSource::locationChanged,
-                                          this, [this](double latitude, double longitude, double altitude)
-    {
-      setLocationInternal(latitude, longitude, altitude);
-    });
+  The default value is \c LocationTrackingMode::Ignore.
+ */
+LocationTrackingMode ArcGISArViewInterface::locationTrackingMode() const
+{
+  return m_locationTrackingMode;
+}
 
-    m_headingChangedConnection = connect(m_locationDataSource, &LocationDataSource::headingChanged,
-                                         this, [this](double heading)
-    {
-      setHeadingInternal(heading);
-    });
+/*!
+  \brief Sets the tracking mode to \a locationTrackingMode.
```

Comment:
```suggestion
  \brief Sets the location tracking mode to \a locationTrackingMode.
```

Reply:
Is locationTrackingMode one possible mode of several "tracking mode"s?  If so, the suggestion may be invalid.

Reply:
No, for the moment, there is only one "tracking mode". But in the future, maybe more tracking mode (for mobile device's accelerometers for example) can be added.